src/main/kotlin/dev/stalla/parser/namespace/PodcastindexParser.kt

```diff
@@ -54,7 +55,7 @@ internal object PodcastindexParser : NamespaceParser() {
         val owner = getAttributeByName("owner")?.value.trimmedOrNullIfBlank()
         val locked = textAsBooleanOrNull()
 
-        if (owner == null || locked == null) return null
+        if (!allNotNull(owner, locked)) return null
```

Comment:
So this is a tricky one here. Initially I used `allNull` here (it's the obvious choice right?). But from `allNull` we don't get the smart cast we want for the return below, because if `allNull` is `false` we only know that some of the arguments are not `null`, but no guarantee for all of them. Only when `!allNotNull(...) == false` (= all arguments are not null) is true, the compiler can do the desired smart cast.

Reply:
Dang, you're right. Well, let's keep it as is now, but would be great if we eventually come up with a nicer way to express the contract. We can hope...